config/nrfconnect/chip-module/CMakeLists.txt

```diff
@@ -222,6 +222,7 @@ chip_gn_arg_bool  ("chip_detail_logging"                    CONFIG_MATTER_LOG_LE
 chip_gn_arg_bool  ("chip_automation_logging"                "false")
 chip_gn_arg_bool  ("chip_malloc_sys_heap"                   CONFIG_CHIP_MALLOC_SYS_HEAP)
 chip_gn_arg_bool  ("chip_enable_wifi"                       CONFIG_WIFI_NRF700X)
+chip_gn_arg_bool  ("chip_multi_variant_ota"                 CONFIG_THREAD_WIFI_SWITCHING)
```

Comment:
That's a minor, but maybe it would be cleaner to create a dedicated Kconfig? I would imagine it could be CONFIG_MULTI_VARIANT_OTA and then imply it in the CONFIG_THREAD_WIFI_SWITCHING.

Reply:
IMHO the clean solution would be simply to support OTA whenever the CHIP_OTA_REQUESTOR is set, regardless if we have single or multiple variant application. This is actually the current state. On the other hand I agree that mapping the CONFIG_THREAD_WIFI_SWITCHING into chip_multi_variant_ota does not look clean... but @kkasperczyk-no do you have any idea how to make it more neat without introducing another Kconfig (which preferably I would like to avoid)? I mean we may just build all OTA related files but that's rather a no-go considering our flash limitations.

Reply:
But why would you like to avoid it? I know it may seem we have already bunch of kconfig options, but the fact is that our platform is very complex and this is the most effective way of configuring our implementation to disable certain parts of a code. 
Honestly I thought this functionality will be used only by the switched sample, so we will somehow implement it in the sample code to avoid making our platform implementation even more complex with another variant of it. Nevertheless I understand it is much more convenient to do that in the platform. In such case I would recommend wrap it into new kconfig option to make some new functionality clearly separated from the single sample.